File: coding/08th October 2025/Replace Element with Greatest Element on Right side.java
//Replace Element with Greatest Element on Right side
class Solution 
{
    public int[] replaceElements(int[] arr) 
    {
        int n = arr.length;
        int[] res = new int[n];
        int max = arr[n - 1];
        for(int i = n - 2; i >= 0; i--)
        {
            res[i] = max;
            if(arr[i] > max)
                max = arr[i];
        }
        res[n - 1] = -1;
        return res;
    }
}

File: Coding in Python/27th December 2025/Replace Element with Greatest Element on Right side.py
#Replace Element with Greatest Element on Right side
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        n = len(arr)
        res = [0] * n
        res[n - 1] = -1
        max = arr[n - 1]
        for i in range(n - 2, -1, -1):
            res[i] = max
            if arr[i] > max:
                max = arr[i]
        return res
